### DISCUSSION TITLE
24 hour clock?

### DISCUSSION THREAD

**Discussion by Skaronator** [dendronhq/dendron#2282] [Q&A]
Hi,

is there any way to display a 24-hour clock on the file header?

![image](https://user-images.githubusercontent.com/2886913/151197426-10a2fe89-47a6-451b-bc02-2ffb30a472a8.png)

**Comment by cconrad** [comment]
Hi @skaronator,

24-hour clock works for me. What are the regional /locale settings in your operating system?

**Reply by Skaronator** [reply]
Good to know that there is support for 24-hours. 

I'm on Pop OS 21.10 (its a ubuntu fork). The settings looks fine for to me but I can experiment a bit tomorrow:

![image](https://user-images.githubusercontent.com/2886913/151211698-8fc03bc2-2080-415a-9c43-7382e2c567d6.png)
![image](https://user-images.githubusercontent.com/2886913/151211725-cf4be265-690b-4749-8f7d-1d5b4eb642af.png)

**Comment by hikchoi** [comment]
@Skaronator 
This could be configured by setting `defaultTimestampDecorationFormat` to one of the following in your `dendron.code-workspace` file:

- `TIME_24_SIMPLE`
- `TIME_24_WITH_LONG_OFFSET`
- `TIME_24_WITH_SECONDS`
- `TIME_24_WITH_SHORT_OFFSET`

more format options here: https://wiki.dendron.so/notes/19a0ea9d-7292-4a68-bc6f-ffd462a54bc5/#dendrondefaulttimestampdecorationformat

These will follow your locale and format it according to the setting you choose.

**Reply by Skaronator** [reply]
Using the `TIME_24_*` options will only display the time and not the date.
![image](https://user-images.githubusercontent.com/2886913/151346476-333d5d42-b068-47df-a4d1-90943d370063.png)

The default `DATETIME_MED` seems to be already the best option but for whatever reason Dendron detects that I'm on a 12 hour clock instead of 24 hour.

**Reply by hikchoi** [reply]
Internally we are using luxon to format timestamps into locale specific human readable strings. There is no processing that Dendron does here other than passing the timestamp to Luxon.

We could enhance this so that users can provide more options for string formatting in the future, but currently we only support the listed shorthands. Would you mind creating a feature request for this? That way we can track it and also let the community vote for it so we can prioritize this feature in the future. (We do this by asking our community in Discord to vote for a feature to be implemented bi-weekly)

You can submit an issue here: https://github.com/dendronhq/dendron/issues/new?assignees=&labels=&template=feature_request.md&title=
